client/securedrop_client/logic.py

```diff
@@ -1144,3 +1148,10 @@ def update_failed_replies(self) -> None:
         failed_replies = storage.mark_all_pending_drafts_as_failed(self.session)
         for failed_reply in failed_replies:
             self.reply_failed.emit(failed_reply.uuid)
+
+    @pyqtSlot(object)
+    def on_receive_selected_sources(self, sources: set[db.Source]) -> None:
+        self._selected_sources = sources
+
+    def get_selected_sources(self) -> set[db.Source] | None:
+        return self._selected_sources
```

Comment:
Since this is a read-only getter with no side effects or dependencies, why not simplify it to:

```suggestion
    @property
    def selected_sources(self) -> set[db.Source]:
        return self._selected_sources
```

Reply:
Depends on [below](https://github.com/freedomofpress/securedrop-client/pull/2252#discussion_r1814870005) - pending your feedback I can either adjust to instantiate with the empty set, and always return a set, or leave as is.

Reply:
<https://github.com/freedomofpress/securedrop-client/pull/2252#discussion_r1814870005> resolves the typing question.  My other suggestion here was to go from a method called as `.get_selected_sources()` to a `@property` called as `.selected_sources`, but feel free to disregard that too if you feel it's too out of pattern with the rest of the current codebase.